### PR TITLE
test_api_min_max.cpp: use size_t for get_global_id() value

### PR DESCRIPTION
**Pull request by JasonTTang**
In some rare cases where get_global_id() is larger than 2G, the 32bit int type would convert the value into a negative integer.

### CODE REVIEW

**Review by bashbaug — APPROVED** [review]
LGTM!

I'll merge after approval from @jlewis-austin , since he had comments on #1409.

**Review by jlewis-austin — APPROVED** [review]
LGTM!

**Pull request merged**
